examples/agents/ppo-chess-llm.py

```diff
@@ -0,0 +1,572 @@
+        if not isinstance(observation_spec, Composite):
+            raise ValueError(
+                f"observation_spec was expected to be of type Composite. Got {type(observation_spec)} instead."
+            )
+        observation_spec["observation"] = TensorSpec(
+            (), Box(), dtype=torch.bfloat16, device="cuda:7"
+        )
+        return observation_spec
+
+
+class LLMInputTransform(Transform):
+    def __init__(self, san_moves):
+        super().__init__()
+        self.san_moves = san_moves
+
+    def _step(self, tensordict, next_tensordict):
+        legal_move_indices = next_tensordict["legal_moves"]
+        legal_moves = [self.san_moves[i] for i in legal_move_indices if i != 29275]
+        existing_moves = (
+            ", ".join(next_tensordict["san"])
+            if not isinstance(next_tensordict["san"], str)
+            else next_tensordict["san"]
+        )
+        if len(legal_moves) > 0:
+            llm_input = (
+                f"You are playing a game of chess. The list of moves so far are [{existing_moves}] and the legal moves are "
+                f"[{', '.join(legal_moves)}]. Please choose one of the legal moves. "
+                f"Respond only with the following sentence, with no additional explanatory text. Example Answer: I choose {legal_moves[0]}!"
+            )
+        else:
+            llm_input = "The game has ended as there are no legal moves left. "
+        next_tensordict["observation"] = llm_input
+        # print("step", llm_input)
+        return next_tensordict
+
+    def _reset(self, tensordict, tensordict_reset):
+        legal_move_indices = tensordict_reset["legal_moves"]
+        legal_moves = [self.san_moves[i] for i in legal_move_indices if i != 29275]
+        existing_moves = (
+            ", ".join(tensordict_reset["san"])
+            if not isinstance(tensordict_reset["san"], str)
+            else tensordict_reset["san"]
+        )
+        if len(legal_moves) > 0:
+            llm_input = (
+                f"You are playing a game of chess. The list of moves so far are [{existing_moves}] and the legal moves are "
+                f"[{', '.join(legal_moves)}]. Please choose one of the legal moves. "
+                f"Respond only with the following sentence, with no additional explanatory text. Example Answer: I choose {legal_moves[0]}!"
+            )
+        else:
+            llm_input = ""
+        tensordict_reset["observation"] = llm_input
+        # print("reset", llm_input)
+        return tensordict_reset
+
+    def transform_observation_spec(self, observation_spec: Composite):
+        if not isinstance(observation_spec, Composite):
+            raise ValueError(
+                f"observation_spec was expected to be of type Composite. Got {type(observation_spec)} instead."
+            )
+        observation_spec["observation"] = NonTensor(shape=(), example_data="any")
+        return observation_spec
+
+
+def run_player(input_queue, output_queue):
```

Comment:
oh wow that's clever! Didn't think it'd take that to use sunfish
Do you think we should upstream that in ChessEnv?